server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServerResourceManager.java

```diff
@@ -556,14 +557,15 @@ private void manageMemory() {
                 continue;
               }
               Tablet tablet = tabletReport.getTablet();
-              if (!tablet.initiateMinorCompaction(MinorCompactionReason.SYSTEM)) {
-                if (tablet.isClosed()) {
+              var state = context.getTableManager().getTableState(tablet.getExtent().tableId());
+              if (!tablet.initiateMinorCompaction(MinorCompactionReason.SYSTEM, state)) {
+                if (tablet.isClosed() || state.equals(TableState.DELETING)) {
```

Comment:
If a tablet is in the deleteting state for a long time and holding lots of memory, it seems like it could starve other minor compactions from ever running.  Does this deleting of reports help mitigate that?

Reply:
Yeah. I haven't seen the MinC max threads (tserver.compaction.minor.concurrent.max=4) consumed by 4 deleted tablets after removing the report. But I think it is still possible.

I am wondering now if I should move the check to inside of `LargestFirstMemoryManager.tabletsToMinorCompact()` so that the deleted tablets will be skipped and not returned in `tabletsToMinorCompact`. This would allow flushes up to `tserver.compaction.minor.concurrent.max` without having to make another pass. Right now the `tabletsToMinorCompact` won't utilize all of the MinC threads if some tablets are being deleted. If all large tablets are being deleted, then no MinC will run during that pass.

Reply:
I added this change in my latest commits. Let me know what you think.